pyoozie/oozie_api.py

```diff
@@ -0,0 +1,561 @@
+    # ===========================================================================
+
+    def _filter_string(self, type_enum, user=None, name=None, status=None):
+        status_type = self.STATUS_TYPES[type_enum]
+        filters = []
+        if user:
+            filters.append('user={}'.format(user))
+        if name:
+            filters.append('name={}'.format(name))
+        if status:
+            if isinstance(status, status_type):
+                filters.append('status={}'.format(status))
+            else:
+                filters.extend(sorted(['status={}'.format(s) for s in status]))
+        filters = '&filter=' + ';'.join(filters) if filters else ''
+        return filters
+
+    def _jobs_query(self, type_enum, user=None, name=None, status=None, limit=0, details=True):
+        job_type, result_type = self.JOB_TYPE_STRINGS[type_enum]
+        filters = self._filter_string(type_enum, user=user, name=name, status=status)
+        offset = 1
+        chunk = limit if limit else 500
+        jobs = []
+        while True:
+            result = self._get('jobs?jobtype={}{}&offset={}&len={}'.format(job_type, filters, offset, chunk))
+            jobs.extend(result[result_type])
+            offset += chunk
+            if (offset > result['total']) or (limit and offset > limit):
+                break
+
+        if details:
+            return [self.JOB_TYPES[type_enum](self, job).fill_in_details() for job in jobs]
+        else:
+            return [self.JOB_TYPES[type_enum](self, job) for job in jobs]
+
+    def jobs_all_workflows(self, name=None, user=None, limit=0):
+        return self._jobs_query(ArtifactType.Workflow, name=name, user=user, limit=limit)
+
+    def jobs_all_active_workflows(self, user=None):
+        return self._jobs_query(ArtifactType.Workflow, status=Workflow.Status.active(), user=user)
+
+    def jobs_all_running_workflows(self, user=None):
+        return self._jobs_query(ArtifactType.Workflow, status=Workflow.Status.running(), user=user)
+
+    def jobs_running_workflows(self, name, user=None):
+        return self._jobs_query(ArtifactType.Workflow, name=name, status=Workflow.Status.running(), user=user)
+
+    def jobs_last_workflow(self, name, user=None):
+        jobs = self._jobs_query(ArtifactType.Workflow, name=name, user=user, limit=1)
+        if jobs:
+            return jobs[-1]
+        else:
+            raise OozieException.workflow_not_found(name)
+
+    def jobs_workflow_names(self, user=None):
+        jobs = self._jobs_query(ArtifactType.Workflow, user=user, details=False)
+        return set([job.appName for job in jobs])
+
+    def jobs_all_coordinators(self, name=None, user=None, limit=0):
+        return self._jobs_query(ArtifactType.Coordinator, name=name, user=user, limit=limit)
+
+    def jobs_all_active_coordinators(self, user=None):
+        return self._jobs_query(ArtifactType.Coordinator, status=Coordinator.Status.active(), user=user)
```

Comment:
Can we take `Coordinator.Status` out of `Coordinator` and name it `CoordinatorStatus`?